### PR TITLE
bump required version of go

### PR DESCRIPTION
**Pull request by jadekler**
[sort.SliceStable](https://github.com/google/martian/blob/a0d62b8f5af8847d8f523c33f45e509b390383f1/trafficshape/utils.go#L183) and [http.ServeTLS](https://github.com/google/martian/blob/a0d62b8f5af8847d8f523c33f45e509b390383f1/mobile/proxy.go#L249) are go 1.8 only.

The former can be solved with:

```
-               sort.SliceStable(shape.Throttles, func(i, j int) bool {
-                       return shape.Throttles[i].ByteStart < shape.Throttles[j].ByteStart
-                       })
+               sort.Sort(ByByteStart(shape.Throttles))

                defaultBandwidth := DefaultBitrate / 8
                if shape.MaxBandwidth > 0 {
@@ -196,8 +194,19 @@ func parseShapes(ts *Trafficshape) error {
                shape.Actions = append(shape.Actions, throttleActions...)

                // Sort the actions according to their byte offset.
-               sort.SliceStable(shape.Actions, func(i, j int) bool { return shape.Actions[i].getByte() < shape.Actions[j].getByte() })
+               sort.Sort(ByByte(shape.Actions))
        }
        return nil
 }

+type ByByteStart []*Throttle
+
+func (a ByByteStart) Len() int           { return len(a) }
+func (a ByByteStart) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByByteStart) Less(i, j int) bool { return a[i].ByteStart < a[j].ByteStart }
+
+type ByByte []Action
+
+func (a ByByte) Len() int           { return len(a) }
+func (a ByByte) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByByte) Less(i, j int) bool { return a[i].getByte() < a[j].getByte() }
```

The latter is harder to solve. I'm not sure if the intent of this library is to be 1.6 compatible, in which case I'm happy to close this and re-open as an issue. The current case, though, is 1.8+ compatibility, so this PR codifies that.

### CODE REVIEW

**Comment by bramhaghosh** [comment]
Ah yeah - our minimum version is **1.9** (that's when http.ServeTLS is introduced). The documentation that this was 1.6 compatible was out of date an incorrect.

If you could update the travis.yml to remove the 1.8 and update requirements to 1.8, I'm happy to merge this in.

Thanks for the fix, and sorry about the incorrect documentation.

**Comment by jadekler** [comment]
Done! And no prob!

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 language: go
 
 go:
-  - tip
+  - 1.9.x
+  - 1.10.x
```
Comment (hueich) [inline .travis.yml]: You need to put quotes around this, so it doesn't get interpreted as go 1.1.
https://docs.travis-ci.com/user/languages/go/

Alternatively, just remove this line and use 1.9

Comment (jadekler) [reply .travis.yml]: Not true - see the build page for this commit correctly building against 1.9 and 1.10 https://travis-ci.org/google/martian/builds/377094519?utm_source=github_status&utm_medium=notification.

The other projects I contribute similarly use this notation and we've seen no problems with it.

**Pull request merged**